src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -0,0 +1,186 @@
+
+    final static String NO_ACCOUNT_NAME_MESSAGE = "No account name was specified in the request.";
+    private static CType getConfigName() {
+        return CType.INTERNALUSERS;
+    }
+
+    static final List<String> RESTRICTED_FROM_USERNAME = ImmutableList.of(
+            ":" // Not allowed in basic auth, see https://stackoverflow.com/a/33391003/533057
+    );
+
+    @Inject
+    public UserService(
+            ClusterService clusterService,
+            ConfigurationRepository configurationRepository,
+            Settings settings,
+            Client client
+    ) {
+        this.clusterService = clusterService;
+        this.configurationRepository = configurationRepository;
+        this.securityIndex = settings.get(ConfigConstants.SECURITY_CONFIG_INDEX_NAME,
+                ConfigConstants.OPENDISTRO_SECURITY_DEFAULT_CONFIG_INDEX);
+        this.client = client;
+    }
+
+    /**
+     * Load data for a given CType
+     * @param config CType whose data is to be loaded in-memory
+     * @return configuration loaded with given CType data
+     */
+    protected static final SecurityDynamicConfiguration<?> load(final CType config, boolean logComplianceEvent) {
+        SecurityDynamicConfiguration<?> loaded = configurationRepository.getConfigurationsFromIndex(Collections.singleton(config), logComplianceEvent).get(config).deepClone();
+        return DynamicConfigFactory.addStatics(loaded);
+    }
+
+    /**
+     * This function will handle the creation or update of a user account.
+     *
+     * @param accountDetailsAsString A string JSON of different account configurations.
+     * @return InternalUserConfiguration with the new/updated user
+     * @throws UserServiceException
+     * @throws IOException
+     */
+    public SecurityDynamicConfiguration<?> createOrUpdateAccount(String accountDetailsAsString) throws IOException {
+
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode accountDetails = mapper.readTree(accountDetailsAsString);
+        final ObjectNode contentAsNode = (ObjectNode) accountDetails;
+        SecurityJsonNode securityJsonNode = new SecurityJsonNode(contentAsNode);
+
+        final SecurityDynamicConfiguration<?> internalUsersConfiguration = load(getConfigName(), false);
+        String accountName = securityJsonNode.get("name").asString();
+
+        if (accountName == null || accountName.length() == 0) { // Fail if field is present but empty
+            throw new UserServiceException(NO_ACCOUNT_NAME_MESSAGE);
+        }
+
+        if (!securityJsonNode.get("attributes").get("owner").isNull() && !securityJsonNode.get("attributes").get("owner").equals(accountName)) { // If this is a service account
+            verifyServiceAccount(securityJsonNode, accountName);
+            String password = generatePassword();
+            contentAsNode.put("password", password);
```

Comment:
It is not stored, this is pattern for the internalUsers. See below in the code, it is then removed from the config. I don't strictly know that it needs to be added but when it is added we can reuse the existing internal user code below this directly since it expects a password to be provided.

Reply:
Looking this over, I removed it since it was not needed.